baby-gru/src/components/card/MoorhenCarbohydrateCard.tsx

```diff
@@ -0,0 +1,76 @@
+import { moorhen } from "../../types/moorhen";
+import { useSelector } from 'react-redux';
+import {Card, Col, Row } from "react-bootstrap";
+import {useCallback, useEffect, useRef, useState} from "react";
+import { guid } from "../../utils/MoorhenUtils";
+import {PrivateerResultsEntry} from "../../types/privateer";
+
+export const MoorhenCarbohydrateCard = (props: {
+    carbohydrate: PrivateerResultsEntry;
+    molecule: moorhen.Molecule;
+}) => {
+
+    const anchorEl = useRef(null)
+    const width = useSelector((state: moorhen.State) => state.sceneSettings.width)
+
+    const { carbohydrate, molecule } = props
+    const molecules = useSelector((state: any) => state.molecules);
+
+    const handleClick = useCallback(async (e) => {
+        const newCenterString =
+            e.target.dataset.chainid +
+            '/' +
+            e.target.dataset.seqnum +
+            '(' +
+            e.target.dataset.resname +
+            ')';
+
+        if (molecule !== null) {
+            await molecule.centreOn(newCenterString);
+        }
+    }, [molecules]);
+
+    const sugarRef = useCallback(
+        (node: HTMLElement | null) => {
+            if (node !== null) {
+                node.querySelector('svg').style.display = 'block';
+                node.querySelector('svg').style.margin = 'auto';
+
+                const useList = node.querySelectorAll('use');
+
+                for (let i = 0; i < useList.length; i++) {
+                    useList[i].addEventListener('click', handleClick);
+                    useList[i].addEventListener('mousedown', (e) => {
+                        e.stopPropagation();
+                    }, {passive: true});
+                    useList[i].addEventListener('touchstart', (e) => {
+                        e.stopPropagation();
+                    }, {passive: true});
+                }
+            }
+        }, []
+    );
```

Comment:
Have you tested these event listeners are cleared once the component unmounts (I suspect not) ? It is important that they are cleared otherwise these listeners will continue to exist in memory.